{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "84197b04-d9ce-4655-97c5-54bfb9094787",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n#get item\\n*Tasks*\\nFeatures in the node dataset:\\n- MRE and active gene labels (DONE)\\n- Chromosome position (DONE)\\n- Sequence information (DONE)\\n- ChIP-seq signals (DONE)\\n- RNA-seq signals (DONE)\\n- Micro-C signals (for each p-value threshold)\\n    -Un-Filtered\\n    -Interaction Bins\\n    -Contact Count\\n    -Number of Neighbors\\n    \\n    -Filtered\\n    -Interaction Bins\\n    -Contact Count\\n    -Number of Neighbors\\n\\nOutput: datasets are stored as .pkl or .pt files\\n\\nNode DataLoader Features (Chromsome Positions, Sequence Information, RNA-seq signals, ChIP-seq signals, MRE and Gene Labels\\nGraph DataLoader Features (Node Features: Chromosome Positions, Sequence Information, RNA-seq signals, ChIP-seq signals, MRE and Gene Labels; Edge Features: Interaction Bins, Contact Counts, P-Values, Number of Neighbors)\\n-look at types of graph dataloaders\\n\\n*Add Additional Datasets (specify requirements)*\\n\\n#generate a torch dataset (generate node matrix, generate edge matrix - binary from the microC data)\\n'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"\"\"\n",
    "#get item\n",
    "*Tasks*\n",
    "Features in the node dataset:\n",
    "- MRE and active gene labels (DONE)\n",
    "- Chromosome position (DONE)\n",
    "- Sequence information (DONE)\n",
    "- ChIP-seq signals (DONE)\n",
    "- RNA-seq signals (DONE)\n",
    "- Micro-C signals (for each p-value threshold)\n",
    "    -Un-Filtered\n",
    "    -Interaction Bins\n",
    "    -Contact Count\n",
    "    -Number of Neighbors\n",
    "    \n",
    "    -Filtered\n",
    "    -Interaction Bins\n",
    "    -Contact Count\n",
    "    -Number of Neighbors\n",
    "\n",
    "Output: datasets are stored as .pkl or .pt files\n",
    "\n",
    "Node DataLoader Features (Chromsome Positions, Sequence Information, RNA-seq signals, ChIP-seq signals, MRE and Gene Labels\n",
    "Graph DataLoader Features (Node Features: Chromosome Positions, Sequence Information, RNA-seq signals, ChIP-seq signals, MRE and Gene Labels; Edge Features: Interaction Bins, Contact Counts, P-Values, Number of Neighbors)\n",
    "-look at types of graph dataloaders\n",
    "\n",
    "*Add Additional Datasets (specify requirements)*\n",
    "\n",
    "#generate a torch dataset (generate node matrix, generate edge matrix - binary from the microC data)\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3e7dd95b-080e-4733-9fd2-8f55661d2d14",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/users/ssgunase/.local/lib/python3.11/site-packages/pandas/core/arrays/masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).\n",
      "  from pandas.core import (\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a4cd84d2-7d6d-4b5d-a999-b6f191a12389",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "chipseq_dir = '/oscar/data/larschan/shared_data/bind_gps/chipseq'\n",
    "labels_dir = '/oscar/data/larschan/shared_data/bind_gps/labels'\n",
    "data_folder = '/oscar/data/larschan/shared_data/BindGPS/data'\n",
    "#fasta file is here^\n",
    "microC_dir = f'{data_folder}/S2_gfp_RNAi_significant_interactions_v2'\n",
    "\n",
    "resolutions = ['1kb', '5kb']\n",
    "pval_thresholds = ['0_001', '0_01', '0_1']\n",
    "genome = SeqIO.to_dict(SeqIO.parse(f'{data_folder}/dm6.fa', \"fasta\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "febc5012-1b79-4b69-95cc-9d4eaa752589",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#NODE Dataset\n",
    "resolutions = ['1kb', '5kb']\n",
    "for resolution in resolutions:\n",
    "    #rnaseq data\n",
    "    rnaseq_df = pd.read_csv(f'{data_folder}/mE_RNAseq_RPKMnorm.{resolution}_binned.tsv', sep=\"\\t\")\n",
    "    rnaseq_df = rnaseq_df.fillna(0) #replace all NaNs with 0\n",
    "    \n",
    "    rnaseq_df['DNA sequence'] = None\n",
    "    #configure DNA sequence\n",
    "    for index, row in rnaseq_df.iterrows():\n",
    "        chrom, start, end = row['chr'], int(row['start']), int(row['end'])\n",
    "        if chrom in genome:\n",
    "            seq = genome[chrom].seq[start:end]\n",
    "            rnaseq_df.at[index, 'DNA sequence'] = str(seq)\n",
    "\n",
    "    #chipseq data\n",
    "    chipseq_df = pd.read_csv(f'{chipseq_dir}/histone_mods_and_GA_factors.feature_matrix.{resolution}_res.tsv', sep=\"\\t\")  \n",
    "    chipseq_df = chipseq_df.drop(columns=['chr', 'start', 'end'])\n",
    "\n",
    "    #active gene labels\n",
    "    gene_labels_df = pd.read_csv(f'{labels_dir}/dc_active_vs_autosome_active_gene_labels.{resolution}_res.tsv', sep=\"\\t\")\n",
    "    chipseq_df['gene_labels'] = gene_labels_df['labels']\n",
    "\n",
    "    #mre labels\n",
    "    mre_labels_df = pd.read_csv(f'{labels_dir}/autosomal_mre_vs_xchr_mre_labels.{resolution}_res.tsv', sep=\"\\t\")\n",
    "    chipseq_df['mre_labels'] = mre_labels_df['labels']\n",
    "    \n",
    "    node_df = pd.concat([rnaseq_df, chipseq_df], axis=1, join='outer')\n",
    "    #reset the index\n",
    "    node_dropped_df = node_df[node_df['chr'] != 'chrM'] #drop the M chromosome\n",
    "    node_dropped_df = node_dropped_df.reset_index(drop=True)\n",
    "    \n",
    "    dir_path = \"/oscar/data/larschan/shared_data/BindGPS/data/datasets\"\n",
    "    node_dropped_df.to_pickle(f\"{dir_path}/node_dataset_{resolution}.pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "56f5f5f0-bcfa-49fd-bd9b-b45c2512b249",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>counts</th>\n",
       "      <th>expression_level</th>\n",
       "      <th>gene_in_bin</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>DNA sequence</th>\n",
       "      <th>clamp</th>\n",
       "      <th>gaf</th>\n",
       "      <th>...</th>\n",
       "      <th>h3k27me3</th>\n",
       "      <th>h3k36me3</th>\n",
       "      <th>h3k4me1</th>\n",
       "      <th>h3k4me2</th>\n",
       "      <th>h3k4me3</th>\n",
       "      <th>h3k9me3</th>\n",
       "      <th>h4k16ac</th>\n",
       "      <th>psq</th>\n",
       "      <th>gene_labels</th>\n",
       "      <th>mre_labels</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>0</td>\n",
       "      <td>5000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>Cgacaatgcacgacagaggaagcagaacagatatttagattgcctc...</td>\n",
       "      <td>0.011338</td>\n",
       "      <td>0.029695</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>20.003200</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.074890</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>5000</td>\n",
       "      <td>10000</td>\n",
       "      <td>19.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>True</td>\n",
       "      <td>FBgn0002121</td>\n",
       "      <td>TGCCTCTCATTCTGTCTTATTTTACCGCAAACCCAAatcgacaatg...</td>\n",
       "      <td>20.016980</td>\n",
       "      <td>20.031764</td>\n",
       "      <td>...</td>\n",
       "      <td>10.088679</td>\n",
       "      <td>290.071502</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.028571</td>\n",
       "      <td>0.0</td>\n",
       "      <td>70.098778</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>10000</td>\n",
       "      <td>15000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>GAGGAGAATGCAAAAAAGCTAAGAACAAAACAATTACTACAAATCG...</td>\n",
       "      <td>10.023666</td>\n",
       "      <td>10.011295</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>140.053661</td>\n",
       "      <td>0.0</td>\n",
       "      <td>110.071154</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>70.025733</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>15000</td>\n",
       "      <td>20000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>ATTCGACGGCGGTTCTGGGTTATCTATGCTCCAAGTGGCGTATGAA...</td>\n",
       "      <td>0.072610</td>\n",
       "      <td>0.058189</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>200.062504</td>\n",
       "      <td>0.0</td>\n",
       "      <td>130.005405</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>60.001147</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr2L</td>\n",
       "      <td>20000</td>\n",
       "      <td>25000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>FBgn0031209,FBgn0263584</td>\n",
       "      <td>GTGGCCGAATTTATTCTAAACTGAAAATAATAATAAAAATTAATCA...</td>\n",
       "      <td>0.079141</td>\n",
       "      <td>0.071863</td>\n",
       "      <td>...</td>\n",
       "      <td>10.025000</td>\n",
       "      <td>160.083882</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>70.087586</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27508</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3645000</td>\n",
       "      <td>3650000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>True</td>\n",
       "      <td>FBgn0267592</td>\n",
       "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...</td>\n",
       "      <td>0.000641</td>\n",
       "      <td>0.000349</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27509</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3650000</td>\n",
       "      <td>3655000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>GTTCTCCACACAAAAAGAATTTTTTCATATACCCTATATAAACGAA...</td>\n",
       "      <td>0.059530</td>\n",
       "      <td>0.050137</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.009552</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27510</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3655000</td>\n",
       "      <td>3660000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>acacggagtaaaaatccgcccagtttgcttagcctccgccaaacgt...</td>\n",
       "      <td>0.000308</td>\n",
       "      <td>0.000322</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.017110</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27511</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3660000</td>\n",
       "      <td>3665000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...</td>\n",
       "      <td>0.000460</td>\n",
       "      <td>0.000196</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27512</th>\n",
       "      <td>chrY</td>\n",
       "      <td>3665000</td>\n",
       "      <td>3667352</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>AGTACATGGTGTAGCAATTGAAGGTGAAAAGTTAAATGCTGNNNNN...</td>\n",
       "      <td>20.060330</td>\n",
       "      <td>10.072728</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27513 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chr    start      end  counts  expression_level  gene_in_bin  \\\n",
       "0      chr2L        0     5000     0.0               0.0        False   \n",
       "1      chr2L     5000    10000    19.0               2.0         True   \n",
       "2      chr2L    10000    15000     0.0               0.0        False   \n",
       "3      chr2L    15000    20000     0.0               0.0        False   \n",
       "4      chr2L    20000    25000     0.0               1.0         True   \n",
       "...      ...      ...      ...     ...               ...          ...   \n",
       "27508   chrY  3645000  3650000     0.0               1.0         True   \n",
       "27509   chrY  3650000  3655000     0.0               0.0        False   \n",
       "27510   chrY  3655000  3660000     0.0               0.0        False   \n",
       "27511   chrY  3660000  3665000     0.0               0.0        False   \n",
       "27512   chrY  3665000  3667352     0.0               0.0        False   \n",
       "\n",
       "                       gene_id  \\\n",
       "0                            0   \n",
       "1                  FBgn0002121   \n",
       "2                            0   \n",
       "3                            0   \n",
       "4      FBgn0031209,FBgn0263584   \n",
       "...                        ...   \n",
       "27508              FBgn0267592   \n",
       "27509                        0   \n",
       "27510                        0   \n",
       "27511                        0   \n",
       "27512                        0   \n",
       "\n",
       "                                            DNA sequence      clamp  \\\n",
       "0      Cgacaatgcacgacagaggaagcagaacagatatttagattgcctc...   0.011338   \n",
       "1      TGCCTCTCATTCTGTCTTATTTTACCGCAAACCCAAatcgacaatg...  20.016980   \n",
       "2      GAGGAGAATGCAAAAAAGCTAAGAACAAAACAATTACTACAAATCG...  10.023666   \n",
       "3      ATTCGACGGCGGTTCTGGGTTATCTATGCTCCAAGTGGCGTATGAA...   0.072610   \n",
       "4      GTGGCCGAATTTATTCTAAACTGAAAATAATAATAAAAATTAATCA...   0.079141   \n",
       "...                                                  ...        ...   \n",
       "27508  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...   0.000641   \n",
       "27509  GTTCTCCACACAAAAAGAATTTTTTCATATACCCTATATAAACGAA...   0.059530   \n",
       "27510  acacggagtaaaaatccgcccagtttgcttagcctccgccaaacgt...   0.000308   \n",
       "27511  NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN...   0.000460   \n",
       "27512  AGTACATGGTGTAGCAATTGAAGGTGAAAAGTTAAATGCTGNNNNN...  20.060330   \n",
       "\n",
       "             gaf  ...   h3k27me3    h3k36me3  h3k4me1     h3k4me2  h3k4me3  \\\n",
       "0       0.029695  ...   0.000000   20.003200      0.0    0.000000      0.0   \n",
       "1      20.031764  ...  10.088679  290.071502      1.0    0.000000      0.0   \n",
       "2      10.011295  ...   0.000000  140.053661      0.0  110.071154      0.0   \n",
       "3       0.058189  ...   0.000000  200.062504      0.0  130.005405      0.0   \n",
       "4       0.071863  ...  10.025000  160.083882      0.0    0.000000      0.0   \n",
       "...          ...  ...        ...         ...      ...         ...      ...   \n",
       "27508   0.000349  ...   0.000000    0.000000      0.0    0.000000      0.0   \n",
       "27509   0.050137  ...   0.000000    0.000000      0.0    0.000000      0.0   \n",
       "27510   0.000322  ...   0.000000    0.000000      0.0    0.000000      1.0   \n",
       "27511   0.000196  ...   0.000000    0.000000      0.0    0.000000      0.0   \n",
       "27512  10.072728  ...   0.000000    0.000000      0.0    0.000000      0.0   \n",
       "\n",
       "         h3k9me3  h4k16ac        psq  gene_labels  mre_labels  \n",
       "0       0.000000      0.0  30.074890            0           1  \n",
       "1      20.028571      0.0  70.098778            1           1  \n",
       "2       0.000000      0.0  70.025733            1           0  \n",
       "3       0.000000      0.0  60.001147            1           1  \n",
       "4       0.000000      0.0  70.087586            1           0  \n",
       "...          ...      ...        ...          ...         ...  \n",
       "27508   0.000000      0.0   0.000000            0           0  \n",
       "27509   0.000000      0.0  10.009552            0           1  \n",
       "27510   0.000000      0.0   0.017110            0           1  \n",
       "27511   0.000000      0.0   0.000000            0           0  \n",
       "27512   0.000000      0.0   0.000000            0           0  \n",
       "\n",
       "[27513 rows x 21 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# node_dropped_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6ac037ba-aeb5-4911-adca-7ef54fccb606",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "#EDGE Dataset\n",
    "# chromosome M is not in there\n",
    "resolutions = ['1kb', '5kb']\n",
    "pval_thresholds = ['0_001', '0_01', '0_1']\n",
    "for resolution in resolutions:\n",
    "    for pval in pval_thresholds:\n",
    "        microc_df = pd.read_table(f'{microC_dir}/S2_gfpi_loops.{resolution}_res.p{pval}.v2.bedpe')\n",
    "        edges = microc_df.loc[:,'bin1':'bin2']\n",
    "        edge_weight = -1 * np.log10(microc_df['p-value'] + 1e-300)\n",
    "        max = np.percentile(edge_weight, q=99.99) #99% maximum - set that as the maximum to account for outliers\n",
    "        edge_weight = np.minimum(edge_weight, max)\n",
    "        microc_df['p-value_transformed'] = edge_weight\n",
    "\n",
    "        dir_path = \"/oscar/data/larschan/shared_data/BindGPS/data/datasets\"\n",
    "        microc_df.to_pickle(f\"{dir_path}/edge_dataset_{pval}_{resolution}.pkl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e3216659-a3e9-4d3e-98f2-936d01a935ad",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr2L' 'chr2R' 'chr3L' 'chr3R' 'chr4' 'chrX' 'chrY']\n"
     ]
    }
   ],
   "source": [
    "print(microc_df['chr1'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "b7a883c7-7976-4f80-bc4c-2a1bf23e5978",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNAACTCCTCGTTGTATGTTCCGATACAAGCGATTGATGTTTGGTGTCAACGCCGCACCCGAAATCTTTCAACGAAGACTTCAGGAGCTTTATATAGCGTGTTCTAATGTCTTGAACTACATTGATGACATCATTGTATTTGGAACAATGAAAATGATCCCGATAACGCGGTCAAGGAAGTCAAAAAGATAGTAAAGGATAATAATATTCTCATGAATGCGGAAAAATGAATTGAGGTTGATACGGAAAAAGTAGCAACAAAAAAGGACCGGAAATTCAATTGGAGTGAAGCAGAGAGAAAAGCTTTCGTGTGTCTTAAGTCTTGTTTTTCAGGATTTGCGACAGATTCCGTTCCCCCAACTGTAATAGCTTGACCGTCCCGGTTGTTCTCCAAAACATTTTTCGTATTACCCTCGCCCGGACCTTAGAAACAATCCGGCCATGGTGTGCGCTAGCAGCGCCAAATAAACATAGAAATCAGTACACAATATAATACATTAATCGATTAAGCTATTTAGCGCAGTACGTCCTCTTCAAGATTTTCAACTGTTTTATGTGTAAATATTTTCCTATATCGATTTTTGAAGAGCGGTATGAATCAGTTTTTTAGCTTTATCTTGGGTACTTAAATTAAGATAGACAAAACTAATAATTATTTTAAATATTTATTGCAGGTTGTGGTTTGTGAACTTAACCGACGGTATTGCAATCAGCATGGCTCG\n"
     ]
    }
   ],
   "source": [
    "seq = genome['chrY'].seq[3663000:3664000] #look into this\n",
    "print(seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "83199d01-5065-45ae-b7ce-b9b2f1df6cb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# unique_chr = node_df['chr'].unique()\n",
    "# unique_chr"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
